File: Algebra_ImproperToMixed.java
package algebra;

import java.util.List;

public class Algebra {

    public static void main(String[] args) {

        int numerator;
        int denominator;
        int whole_number;
        int[] solution;

        System.out.println("Welcome to our demonstration sequels");
        System.out.println("Hope you enjoy (and follow) the lessons.");
        System.out.println("");
        
        /*
        * Convert fractions from Improper to Mixed
         */
        numerator = 10;
        denominator = 3;

        System.out.println("Converting from Improper to Mixed the fraction:");
        // Print as fraction
        System.out.printf("%55d%n", numerator);
        System.out.printf("%54s%n", "―");
        System.out.printf("%55d%n", denominator);

        // use the ImproperToMixed class
        ImproperToMixed imp_mix = new ImproperToMixed(numerator, denominator);
        solution = imp_mix.doConvert();
        whole_number = solution[0];
        numerator = solution[1];

        System.out.println();

        System.out.printf("%52d%n", numerator);
        System.out.printf("%50s%d%s%n", "Answer =  ", whole_number, "―");
        System.out.printf("%52d%n", denominator);

        System.out.println("\n\n");
    }

}

File: Algebra_ImproperToMixed.cpp
// Algebra.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include "ImproperToMixed.h"

#include <iostream>
#include <exception>

using namespace std;

int main()
{
	cout << "\n    Welcome to our demonstration sequels";
	cout << "\n    Hope you enjoy (and follow) the lessons.\n\n";

	unsigned int numerator;
	unsigned int denominator;
	unsigned int whole_number;

	try {
		/*
		* Convert fractions from Improper to Mixed
		*/
		numerator = 10;
		denominator = 3;

		cout << "\n    Converting from Improper to Mixed the fraction:\n";
		// Print as fraction
		printf("%56u\n", numerator);
		printf("%56s\n", "-");
		printf("%56u\n", denominator);

		// use the ImproperToMixed class
		ImproperToMixed imp_mix(numerator, denominator);
		imp_mix.doConvert();

		printf("\n%52u\n", imp_mix.new_numerator);
		printf("%50s%u%s\n", "Answer =   ", imp_mix.whole_number, "-");
		printf("%52u\n", denominator);

		cout << "\n\n";

	} catch (exception& e) {
		cout << "\n" << e.what() << "\n";
	}

    return 0;
}

